source/connection-monitoring-and-pooling/tests/pool-checkout-minSize-connection-maxConnecting.yml

```diff
@@ -0,0 +1,63 @@
+version: 1
+style: integration
+description: threads blocked by maxConnecting check out minSize connections
+runOn:
+  -
+    # required for blockConnection in fail point
+    minServerVersion: "4.4.0"
+failPoint:
+  configureFailPoint: failCommand
+  mode: "alwaysOn"
+  data:
+    failCommands: ["isMaster","hello"]
+    closeConnection: false
+    blockConnection: true
+    blockTimeMS: 500
+poolOptions:
+  # allows both thread1 and the background thread to start opening connections concurrently
+  minPoolSize: 2
+  # gives opportunity for the checkout in thread2 to open a new connection, which it must not do nonetheless
+  maxPoolSize: 3
+  waitQueueTimeoutMS: 5000
```

Comment:
Do we need backgroundThreadIntervalMS here?

Reply:
No, all we need is for the BG thread to start a single [run](https://github.com/mongodb/specifications/blob/master/source/connection-monitoring-and-pooling/connection-monitoring-and-pooling.rst#background-thread), which it does without artificial delays, and then we do not care when the next run starts.

Reply:
It is not explicitly specified that the background thread should be started ASAP ("immediately" is not an appropriate term here), but the specification of `minPoolSize` (which is unachievable, and, therefore, should have been formulated differently) implies this: [_"The minimum number of Connections that MUST exist at any moment in a single connection pool."_](https://github.com/mongodb/specifications/blob/master/source/connection-monitoring-and-pooling/connection-monitoring-and-pooling.rst#id1)